scipy/signal/tests/test_signaltools.py

```diff
@@ -2800,6 +2800,188 @@ def test_hilbert2_types(self, dtype):
+                          msg="Residual calculation error (residual='all')")
+
+        if bp_in[1] is not None:
+            Zr_desired[bp_in[1]:] = 0
+        self.assert_close(Ze2_lp, Ze2_desired,
+                          msg="Envelope calculation error (residual='lowpass')")
+        self.assert_close(Zr_lp, Zr_desired,
+                          msg="Residual calculation error (residual='lowpass')")
+
+    @pytest.mark.parametrize(
+        "               Z,        bp_in,         Ze2_desired,         Zr_desired",
+        [([0, 5, 0, 5, 0], (None, None),    [5, 0, 10, 0, 5],    [0, 0, 0, 0, 0]),
+         ([1, 5, 0, 5, 2],      (-1, 2),    [5, 0, 10, 0, 5],    [1, 0, 0, 0, 2]),
+         ([1, 2, 6, 0, 6, 3],   (-1, 2), [0, 6, 0, 12, 0, 6], [1, 2, 0, 0, 0, 3])
+         ])
+    def test_envelope_complex_signals(self, Z, bp_in, Ze2_desired, Zr_desired):
+        """Test envelope calculation with complex-valued test signals.
+
+        We only need to test for the complex envelope here, since the ``Nones``s in the
+        bandpass filter were already tested in the previous test.
+        """
+        z = sp_fft.ifft(sp_fft.ifftshift(Z))
+        ze2, zr = envelope(z, bp_in, residual='all', squared=True)
+        Ze2, Zr = (sp_fft.fftshift(sp_fft.fft(z_)) for z_ in (ze2, zr))
+
+        self.assert_close(Ze2, np.array(Ze2_desired).astype(complex),
+                          msg="Envelope calculation error")
+        self.assert_close(Zr, np.array(Zr_desired).astype(complex),
+                          msg="Residual calculation error")
+
+    def test_envelope_verify_axis_parameter(self):
+        """Test for multi-channel envelope calculations. """
+        z = sp_fft.irfft([[1, 0, 2, 2, 0], [7, 0, 4, 4, 0]])
+        Ze2_desired = np.array([[4, 2, 0, 0, 0], [16, 8, 0, 0, 0]],
+                               dtype=complex)
+        Zr_desired = np.array([[1, 0, 0, 0, 0], [7, 0, 0, 0, 0]], dtype=complex)
+
+        ze2, zr = envelope(z, squared=True, axis=1)
+        ye2T, yrT = envelope(z.T, squared=True, axis=0)
+        Ze2, Ye2, Zr, Yr = (sp_fft.rfft(z_) for z_ in (ze2, ye2T.T, zr, yrT.T))
+
+        self.assert_close(Ze2, Ze2_desired, msg="2d envelope calculation error")
+        self.assert_close(Zr, Zr_desired,  msg="2d residual calculation error")
+        self.assert_close(Ye2, Ze2_desired, msg="Transposed 2d envelope calc. error")
+        self.assert_close(Yr, Zr_desired, msg="Transposed 2d residual calc. error")
+
+    def test_envelope_verify_axis_parameter_complex(self):
+        """Test for multi-channel envelope calculations with complex values. """
+        z = sp_fft.ifft(sp_fft.ifftshift([[1, 5, 0, 5, 2], [1, 10, 0, 10, 2]], axes=1))
+        Ze2_des = np.array([[5, 0, 10, 0, 5], [20, 0, 40, 0, 20],],
+                           dtype=complex)
+        Zr_des = np.array([[1, 0, 0, 0, 2], [1, 0, 0, 0, 2]], dtype=complex)
+
+        kw = dict(bp_in=(-1, 2), residual='all', squared=True)
```

Comment:
Can you also test for squared = False?

Reply:
Consult [comment above](https://github.com/scipy/scipy/pull/20097/files#r1694233553).